roachtest: fix acceptance tests owner tags

RaduBerinde opened the pull request:
When we `Add` a test to the registry, an `owner-XX` tag is added. For the acceptance tests, we were aliasing the tags between all tests (effectively, all tests get all owner tags).

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110679)
<!-- Reviewable:end -->

renatolabs commented on pkg/cmd/roachtest/tests/acceptance.go:
```diff
@@ -97,15 +85,25 @@ func registerAcceptance(r registry.Registry) {
+				// blank issue for the "acceptance" parent test. Make sure to
+				// teach that script (if it's still used at that point) should
+				// this naming scheme ever change (or issues such as #33519)
+				// will be posted.
```
Since we're here, do you mind deleting this entire comment? It no longer applies.

renatolabs reviewed (approved):
Thank you!

RaduBerinde committed:
roachtest: fix acceptance tests owner tags

When we `Add` a test to the registry, an `owner-XX` tag is added. For
the acceptance tests, we were aliasing the tags between all tests
(effectively, all tests get all owner tags).

Epic: none
Release note: None

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110679#-:-NeOSXTAASGcqPfp6CoN:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @rachitgsrivastava, and @renatolabs)

___
*[`pkg/cmd/roachtest/tests/acceptance.go` line 94 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/110679#gh-1327420364:-NeOShxrEiRZo8QVdbfD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b456bcc27fb154ed14a2e9518c2f857ad9ba7134/pkg/cmd/roachtest/tests/acceptance.go#L94)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Since we're here, do you mind deleting this entire comment? It no longer applies.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+

srosenberg commented:
blathers backport 22.2 23.1
